src/freeze-dry/common.test.js

```diff
@@ -0,0 +1,97 @@
+            '<html><head></head><body></body></html>',
+            'text/html'
+        )
+        removeNode(rootElement.querySelector('head'))
+        expect(rootElement.querySelector('head')).toBeNull()
+    })
+})
+
+describe('urlToDataUri', () => {
+    test('should return a valid dataUri for a URL', async () => {
+        const spy = jest.spyOn(responseToDataUri, 'default')
+        await urlToDataUri('https://example.com/page')
+        expect(spy).toHaveBeenCalled()
+        spy.mockRestore()
+    })
+
+    test('should return a "about:invalid" for a invalid URL', async () => {
+        expect.assertions(2)
+        const spy = jest.spyOn(responseToDataUri, 'default').mockImplementation(() => {
+            throw new Error('mock error')
+        })
+        const dataUri = await urlToDataUri('example.com')
+        expect(spy).toHaveBeenCalled()
+        expect(dataUri).toBe('about:invalid')
+        spy.mockRestore()
+    })
+
+    test('should return a "about:invalid" for an error URL', async () => {
+        fetch.mockResponseOnce(new Error())
+        const dataUri = await urlToDataUri('example.com')
+        expect(dataUri).toBe('about:invalid')
+    })
+
+    test('should return the same url if already in dataUri form', async () => {
+        const dataUri = await urlToDataUri('data:;base64,test')
+        expect(dataUri).toBe('data:;base64,')
+    })
+})
+
+describe('inlineUrlsInAttributes', () => {
+    const docUrl = 'https://example.com/page'
+    const parser = new DOMParser()
+
+    test('should change the URL in <img> tag to a dataUri', async () => {
+        fetch.mockResponseOnce(new Blob([ {example: 'Lorem Ipsum'} ]))
+        expect.assertions(2)
+        const rootElement = parser.parseFromString(
+            '<html><body><img src="public/image/background.png" alt="background" /></body></html>',
+            'text/html'
+        )
+        await inlineUrlsInAttributes({elements: 'img', attributes: 'src', fixIntegrity: true, rootElement, docUrl})
+        expect(rootElement.querySelector('img').getAttribute('data-original-src')).toBe('public/image/background.png')
+        expect(rootElement.querySelector('img').getAttribute('src')).toBe('data:;base64,W29iamVjdCBPYmplY3Rd')
+    })
+
+    test('should change the URL in the <link> tag to a dataUri', async () => {
+        fetch.mockResponseOnce(new Blob([ {example: 'Lorem Ipsum'} ]))
+        expect.assertions(2)
+        const rootElement = parser.parseFromString(
+            '<html><head><link rel="icon" href="public/image/favicon.ico"></head></html>',
+            'text/html'
+        )
+        await inlineUrlsInAttributes({elements: 'link', attributes: 'href', fixIntegrity: true, rootElement, docUrl})
+        expect(rootElement.querySelector('link').getAttribute('data-original-href')).toBe('public/image/favicon.ico')
+        expect(rootElement.querySelector('link').getAttribute('href')).toBe('data:;base64,W29iamVjdCBPYmplY3Rd')
+    })
```

Comment:
Is it worth adding this test? It looks like only the tag name is different (`img` → `link`). We could end up adding tests for every element type. ;)